tap_appstore/__init__.py

```diff
@@ -1,30 +1,53 @@
 #!/usr/bin/env python3
+import atexit
 import json
+import os
+import tempfile
 
 import singer
 from appstoreconnect import Api
-from singer import utils, Catalog
+from singer import Catalog, utils
 
 from tap_appstore.discover import discover, do_discover
 from tap_appstore.sync import sync
 
 REQUIRED_CONFIG_KEYS = [
     'key_id',
-    'key_file',
     'issuer_id',
     'vendor',
     'start_date'
 ]
 
 LOGGER = singer.get_logger()
 
+def _delete_temp_file(temp_filename) -> None:
+    try:
+        LOGGER.info(f"Cleaning up: {temp_filename}")
+    finally:
+        os.remove(temp_filename)
+        LOGGER.info(f"Temporary key_file deleted: {temp_filename}")
+
+def _create_temp_key_file(key_file_str) -> str:
+    temp_file = tempfile.NamedTemporaryFile(delete=False)
+    temp_filename = temp_file.name
+    with open(temp_filename, "w") as file:
+        file.write(key_file_str)
```

Comment:
if `key_file_str` is read in from the environment, the newline characters can get escaped from `\n` to `\\n`. This would later cause the private key to get mangled.

one strategy could be to replace `file.write(key_file_str)` with `file.write(key_file_str.replace(r'\n', '\n'))`. This would not impact any strings that don't undergo escaping.

source: https://stackoverflow.com/questions/48505964/reading-in-quoted-environment-variable-new-line

Reply:
I tested the abovementioned strategy locally (setting the environment variable on a docker container, and using that to invoke the tap via meltano) and it worked like a charm.

Reply:
hmm we also use this in meltano in its current state with out any problems. I wonder whats different 🤔

Reply:
i can share a bit about my testing setup.

initially, i had run your branch on my local machine in meltano docker container with the tap's config looking like this
```
        vendor: "1234"
        issuer_id: "daklj1234j"
        key_file_str: "----- begin private key ----... ---- end private key"
        key_id: "abcd-efgh-..."
        start_date: "yyyy-mm-ddThh:mm:ssZ"
```

this worked at first.

however after putting the secrets into a vault & setting it up so that the secrets are fetched from the vault and then passed to the docker container, it stopped working.

i commented out the bit of code that deletes the temp file, went inside the container & looked at the temp file. I found that the newlines were mangled due to an extra escape

asking around, one of my co-workers suggested encoding the env variable to base64 and decoding them before the application uses it and I also found the stackoverflow post linked in earlier comment.

ultimately felt the `replace(r'\n', '\n')` was simpler to reason about and implement.

Reply:
i did some further testing and the env var is respecting the new line. 
it does feel like fix should be applied on the vault code itself, rather than within tap-appstore.